ocaml/xapi/xapi_xenops.ml

```diff
@@ -613,6 +613,19 @@ module MD = struct
       (List.combine (Range.to_list (Range.make 0 (List.length devs))) devs)
 
   let get_target_pci_address ~__context vgpu =
+    (* Reallocate a vgpu to during vm checkpoint *)
+    let vm = vgpu.Db_actions.vGPU_VM in
+    let implementation = Db.VGPU_type.get_implementation ~__context ~self:vgpu.Db_actions.vGPU_type in
+    let vgpu =
+      match Db.VM.get_power_state ~__context ~self:vm, implementation with
+      | `Suspended, `nvidia ->
```

Comment:
What happens if it is not Nvidia, but AMD or Intel? Can we support suspend on them with the same code?

Reply:
1. For this function `get_target_pci_address`, if AMD or Intel, then the record of vgpu will be the original one, nothing changes with the old logic.
2. For AMD or Intel vGPU vms suspendable, please see https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi_vm_lifecycle.ml#L265, only nvidia vGPU support suspend and checkpoint(the newly added).

Thanks, Edvin.